prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -370,6 +409,19 @@ class ActionHistory {
         reportFile.itemCount = report.itemCount
         reportsOut[reportFile.reportId] = reportFile
         filteredReportRows[reportFile.reportId] = report.filteringResults
+        report.filteringResults.forEach {
+            trackDetails(
+                ActionDetail(
+                    ActionDetail.DetailScope.report,
+                    "",
+                    it,
+                    reportId = report.id,
+                    action = action,
+                    type = ActionDetail.Type.filter,
+                )
+            )
+        }
```

Comment:
This feels very clunky, is there somewhere else in the code this should be happening?